gnucash/import-export/csv-imp/gnc-csv-price-import-settings.cpp

```diff
@@ -0,0 +1,258 @@
+/*******************************************************************\
+ * gnc-csv-price-import-settings.cpp -- Price CSV Import Settings   *
+ *                                                                  *
+ * Copyright (C) 2017 Robert Fewell                                 *
+ *                                                                  *
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+\********************************************************************/
+/** @file gnc-csv-price-import-settings.cpp
+    @brief CSV Import Settings
+    @author Copyright (c) 2014 Robert Fewell
+    @author Copyright (c) 2016 Geert Janssens
+*/
+
+#include "gnc-csv-import-settings.hpp"
+#include "gnc-csv-price-import-settings.hpp"
+#include <sstream>
+
+extern "C"
+{
+#include <config.h>
+
+#include <gtk/gtk.h>
+#include <glib/gi18n.h>
+
+#include "Account.h"
+#include "gnc-state.h"
+#include "gnc-ui-util.h"
+}
+
+const std::string settings_type{"PRICE"};
+
+#define CSV_COL_TYPES    "ColumnTypes"
+
+#define CSV_TO_CURR      "PriceToCurrency"
+#define CSV_FROM_COMM    "PriceFromCommodity"
+
+G_GNUC_UNUSED static QofLogModule log_module = GNC_MOD_IMPORT;
+
+preset_vec_price presets_price;
+
+static std::shared_ptr<CsvPriceImpSettings> create_int_no_preset(void)
```

Comment:
Aside the template specialization this function is identical across settings types. So it would make sense to share the code as well. One way to do this would be to add the function as a templatized, static, protected class member to CsvImportSettings:
```
template<class SType>static std::shared_ptr<CsvImportSettings> create_no_preset()
```
I'm not 100% sure of the location of the static keyword. Note also the generic class type I'm returning rather than the specific one per import type (see other comments for the reasoning).

As your get_import_presets_* functions need to be friends of that class anyway, they can still call this common function as
```
CsvImportSettings::create_no_preset<CsvPriceImpSettings>()
```
for CsvTransImpSettings you only need to change the template specialization to <CsvTransImpSettings>